Transformations/TransformEigen/TransformEigen.cpp

```diff
@@ -0,0 +1,257 @@
+/*
+Convenience functions for pose and co-ordinate system transformations.
+*/
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+
+#include <opencv2/core/core.hpp>
+
+#include <cmath>
+
+#include "TransformEigen.h"
+#include <iostream>
+
+namespace TransformUtils
+{
+    enum Rotation
+    {
+        INTRINSIC,
+        EXTRINSIC
+    };
+
+    cv::Mat eigen2cv(const Eigen::MatrixXd &eigenMat)
+    {
+        cv::Mat cvMat(static_cast<int>(eigenMat.rows()), static_cast<int>(eigenMat.cols()), CV_64FC1, cv::Scalar(0));
+
+        for(int i = 0; i < cvMat.rows; i++)
+        {
+            for(int j = 0; j < cvMat.cols; j++)
+            {
+                cvMat.at<double>(i, j) = eigenMat(i, j);
+            }
+        }
+
+        return cvMat;
+    }
+
+    Eigen::MatrixXd cv2eigen(const cv::Mat &cvMat)
+    {
+        if(cvMat.dims > 2)
+        {
+            throw "Invalid matrix dimensions. Expected 2D.";
+        }
+
+        Eigen::MatrixXd eigenMat(cvMat.rows, cvMat.cols);
+
+        for(int i = 0; i < cvMat.rows; i++)
+        {
+            for(int j = 0; j < cvMat.cols; j++)
+            {
+                eigenMat(i, j) = cvMat.at<double>(i, j);
+            }
+        }
+
+        return eigenMat;
+    }
+
+    Eigen::Vector4d reorderQuaternion(Eigen::Quaterniond &quaternion)
+    {
+        Eigen::Vector4d reorderedQuaternion;
+
+        reorderedQuaternion[0] = quaternion.coeffs()[3];
+        for(int i = 0; i < 3; i++)
+            reorderedQuaternion[i + 1] = quaternion.coeffs()[i];
+
+        return reorderedQuaternion;
+    }
+
+    Eigen::Vector4d reorderAxisAngle(Eigen::AngleAxisd &axisAngle)
+    {
+        Eigen::Vector4d reorderedAxisAngle;
+
+        for(int i = 0; i < 3; i++)
+            reorderedAxisAngle[i] = axisAngle.axis()[i];
+        reorderedAxisAngle[3] = axisAngle.angle();
+
+        return reorderedAxisAngle;
+    }
+
+    Eigen::Array3d rotmat2rpy(Eigen::Matrix3d rotationMatrix, std::string orientation = "xyz")
+    {
+        char a[3];
+        for(int i = 0; i < 3; i++)
+        {
+            switch(orientation[i])
+            {
+                case 'x': a[i] = 0; break;
+                case 'y': a[i] = 1; break;
+                case 'z': a[i] = 2; break;
+                default: throw "Invalid orientation";
+            }
+        }
+
+        Eigen::Array3d rollPitchYaw = rotationMatrix.eulerAngles(a[0], a[1], a[2]);
```

Comment:
@torbsorb - The function .eulerAngles works only for zxz. Did you miss that or am I missing something?